internal/init/init.go

```diff
@@ -21,12 +21,11 @@ type initOpts struct {
 	Directory      string `opts:"help=output directory"`
 }
 
-func Register(parent opts.Opts) {
-	in := initOpts{
+func New() opts.Opts {
```

Comment:
Can I suggest `func New() opts.SubOpts` so that unnecessary/illegal signatures are available eg EmbedGlobalFlagSet, 	Complete, EmbedFlagSet ...

AddCommand should only take a SubOpts.

I know, I know we now need a opts.NewSub()